src/transformers/modeling_utils.py

```diff
@@ -2551,24 +2551,6 @@ def save_pretrained(
             if not _hf_peft_config_loaded:
                 model_to_save.config.save_pretrained(save_directory)
             if self.can_generate():
-                # generation config built from the model config + the model config holds generation kwargs -> generate
```

Comment:
NOTE: the check removed here is redundant with the check that gets upgraded from warning to exception in this PR. We call `config.save_pretrained()` above in L2552, which runs the upgraded check 👼